tests/utils.test.ts

```diff
@@ -0,0 +1,50 @@
+import { validateIdentityName } from '../src/utils'
+
+describe(validateIdentityName.name, () => {
+  it('returns error state when string less than 8 characters', () => {
+    const result = validateIdentityName('john')
+    expect(result).toEqual('error_minimum_length')
+  })
+
+  it('returns error state when using uppercase chars', () => {
+    const result = validateIdentityName('COOLNAMEBRO')
+    expect(result).toEqual('error_illegal_character')
+  })
+
+  it('returns error state when using non-underscore symbols', () => {
+    const result = validateIdentityName('kyranj@mie')
+    expect(result).toEqual('error_illegal_character')
+  })
+
+  it('returns error state when using non-underscore symbols', () => {
+    const result = validateIdentityName('COOLNAMEBRO')
+    expect(result).toEqual('error_illegal_character')
+  })
+
+  it('returns error state when using sneaky homoglyphs', () => {
+    const legitIndentity = 'kyranjamie'
+    const homoglyph = 'kyrаnjamie'
+    // eslint-disable-next-line 
+    // @ts-ignore
+    expect(legitIndentity === homoglyph).toEqual(false)
+    const shouldPassResult = validateIdentityName(legitIndentity)
+    expect(shouldPassResult).toBeNull()
+
+    const shouldFailResult = validateIdentityName(homoglyph)
+    expect(shouldFailResult).toEqual('error_illegal_character')
+  })
+
+  it('allows a selection of legit names', () => {
+    const names = [
+      'kyranjamie',
+      'jasperjansz101',
+      'auln3auuuu',
+      'honeypot1337',
+      'marina_p',
+      '42214218',
+      'emotion_trouble_solution_juice',
+      '________'
```

Comment:
I understand this may be allowed per the browser's rules, but I think we should update it so that:

- You can't start with an underscore
- You can't have more than one underscore in a row

Reply:
I recommend we don't apply any restrictions here as far as underscores go since if the user wants to confuse their username, we should let them.